File: typescript/tracing/langchain/02_chat_model_stream.ts
import { fakeChat, initRespan, messageText, shutdown, tracingConfig } from "./_shared";

export async function chatModelStream(): Promise<void> {
  const runtime = await initRespan("typescript-langchain-chat-model-stream");
  const model = fakeChat(["Streaming hello from LangChain."]);

  try {
    const chunks = await model.stream(
      "Stream a short hello.",
      tracingConfig(runtime, "chat_model_stream"),
    );
    const parts: string[] = [];
    for await (const chunk of chunks) {
      parts.push(messageText(chunk));
    }
    console.log(parts.join(""));
  } finally {
    await shutdown(runtime);
  }
}

if (import.meta.url === `file://${process.argv[1]}`) {
  await chatModelStream();
}

File: python/tracing/langchain/02_chat_model_stream.py
"""Chat model stream."""

from langchain_core.language_models.fake_chat_models import FakeListChatModel

from _shared import flush, init_telemetry, message_text, tracing_config


def chat_model_stream() -> None:
    telemetry = init_telemetry("langchain-chat-model-stream")
    model = FakeListChatModel(responses=["Streaming chat output."])
    try:
        chunks = [
            message_text(chunk)
            for chunk in model.stream(
                "Stream a short status update.",
                config=tracing_config("chat_model_stream"),
            )
        ]
        print("".join(chunks))
    finally:
        flush(telemetry)


if __name__ == "__main__":
    chat_model_stream()
